pkg/signup/service/signup_service.go

```diff
@@ -223,12 +233,12 @@ func (s *ServiceImpl) reactivateUserSignup(ctx *gin.Context, existing *toolchain
 // GetSignup returns Signup resource which represents the corresponding K8s UserSignup
 // and MasterUserRecord resources in the host cluster.
 // Returns nil, nil if the UserSignup resource is not found or if it's deactivated.
-func (s *ServiceImpl) GetSignup(userID string) (*signup.Signup, error) {
-
-	// Retrieve UserSignup resource from the host cluster
-	userSignup, err := s.CRTClient().V1Alpha1().UserSignups().Get(EncodeUserID(userID))
-	if err != nil {
-		if apierrors.IsNotFound(err) {
+func (s *ServiceImpl) GetSignup(userID, username string) (*signup.Signup, error) {
+	// Retrieve UserSignup resource from the host cluster, using the specified UserID and username
+	userSignup, err := s.GetUserSignup(userID, username)
+	// If an error was returned, *or* the returned userSignup is nil, then return here
+	if err != nil || userSignup == nil {
+		if err != nil && apierrors.IsNotFound(err) {
 			return nil, nil
```

Comment:
When could happen that the `userSignup` is `nil` and the `err` is not set (is nil as well)? This code is pretty confusing and error-prone.

Reply:
Rewritten to be less confusing.